### DISCUSSION TITLE
Can not download builds under anonymous user when Option "Make builds public" is enabled

### DISCUSSION THREAD

**Discussion by hodev** [codemagic-ci-cd/codemagic-docs#2600] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Build flutter project with android and ios code.
2.  Enable option "Make builds public" in Workflow settings
3. Read text "When you make builds public, all existing and future build logs and build artifacts of this workflow will be publicly available if you share a link to the build or display a build status badge. Direct links to build artifacts and log files will remain accessible even when you disable this option.
Are you sure you wish to continue?" and click  "Yes, make builds public"
4. Open last build and copy link of last build from address bar

### Expected results

1. Open copied link in incognito mode in browser
2. Try to download android or ios artifacts
3. You can download android or ios artifacts

### Actual results

1. Open copied link in incognito mode in browser
2.  Try to download android or ios artifacts
3. You got error "
  "error": "FORBIDDEN",
  "message": "The server could not verify that you are authorized to access the URL requested. You either supplied the wrong credentials (e.g. a bad password), or your browser doesn't understand how to supply the credentials required."
}"

### Build id (optional)

66713bc4086149b3f6e94b6e

**Comment by dam-ease** [comment, marked as answer]
Hello @hodev 
This is the expected behaviour. Toggling the "make builds public" button enables viewing the build publicly but you won't be able to download artifacts via the link.
If you'd like to distribute the artifacts and ensure it is downloadable, kindly check out setting up a [build dashboard](https://docs.codemagic.io/yaml-publishing/build-dashboards/)
